dbt_semantic_interfaces/implementations/semantic_version.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import Optional
+
+from typing_extensions import override
+
+from dbt_semantic_interfaces.implementations.base import (
+    HashableBaseModel,
+    PydanticCustomInputParser,
+    PydanticParseableValueType,
+)
+from dbt_semantic_interfaces.protocols.semantic_version import SemanticVersion
+
+
+class PydanticSemanticVersion(PydanticCustomInputParser, HashableBaseModel):
+    """Pydantic implementation of SemanticVersion."""
+
+    @override
+    def _implements_protocol(self) -> SemanticVersion:
+        return self
+
+    major_version: str
+    minor_version: str
+    patch_version: Optional[str]
+
+    @classmethod
+    def _from_yaml_value(cls, input: PydanticParseableValueType) -> PydanticSemanticVersion:
+        """Parses a MetricInputMeasure from a string (name only) or object (struct spec) input.
+
+        For user input cases, the original YAML spec for a PydanticMetric included measure(s) specified as string names
+        or lists of string names. As such, configs pre-dating the addition of this model type will only provide the
+        base name for this object.
+        """
+        if isinstance(input, str):
+            input_split = input.split(".")
+            if len(input_split) not in (2, 3):
+                raise ValueError(f"Expected version string to be of the form x.y or x.y.z but got {input}")
+            return PydanticSemanticVersion(
+                major_version=input_split[0],
+                minor_version=input_split[1],
+                patch_version=input_split[2] if len(input_split) == 3 else None,
+            )
+        else:
+            raise ValueError(
+                f"{cls.__name__} inputs from YAML files are expected to be of either type string or "
+                f"object (key/value pairs), but got type {type(input)} with value: {input}"
+            )
```

Comment:
We probably want to update this to be a validator similar to https://github.com/dbt-labs/dbt-semantic-interfaces/blob/739fae883f991b92ad6db877223e5068a241947d/dbt_semantic_interfaces/implementations/semantic_manifest.py#L26

Reply:
A little unclear how the validator would work in the case with the custom parser. Mind elaborating?